pyasn1/codec/ber/decoder.py

```diff
@@ -16,35 +19,71 @@
 from pyasn1.type import univ
 from pyasn1.type import useful
 
-__all__ = ['decode']
+
+__all__ = ['decode', 'decodeStream']
 
 LOG = debug.registerLoggee(__name__, flags=debug.DEBUG_DECODER)
 
 noValue = base.noValue
 
 
+def asStream(substrate):
+    # TODO: Apply sparingly or remove
+    # :type substrate: Union[bytes, BytesIO, BufferedReader, OctetString]
+    # :rtype: BufferedReader
+    if isinstance(substrate, univ.OctetString):
+        substrate = substrate.asOctets()
+    if isinstance(substrate, bytes):
+        substrate = BytesIO(substrate)
+    if isinstance(substrate, BytesIO):
+        return BufferedReader(substrate)
+    if isinstance(substrate, BufferedReader):
+        return substrate
+    raise ValueError("%s Cannot be converted to stream" % substrate.__class__)
+
+
+def popSubstream(substrate, length):
+    content = substrate.read(length)
+    return asStream(content)
+
+
+def isEmpty(substrate):
+    # :type substrate: Union[bytes, BytesIO, BufferedReader]
+    # :rtype: bool
+    if isinstance(substrate, bytes):
+        return not bool(substrate)
+    else:
+        return not bool(asStream(substrate).peek(1))
```

Comment:
How about reducing the while function into:

    return not (substrate and asStream(substrate).peek(1))

?